python/ray/util/client/worker.py

```diff
@@ -49,6 +53,21 @@ def __init__(self,
             self.channel = grpc.secure_channel(conn_str, credentials)
         else:
             self.channel = grpc.insecure_channel(conn_str)
+
+        conn_attempts = 0
+        timeout = INITIAL_TIMEOUT_SEC
+        while conn_attempts < connection_retries:
```

Comment:
```suggestion
        while conn_attempts < connection_retries + 1:
```

To avoid confusion around the zero retries case.